StringCalculator/StringCalculator.cs

```diff
@@ -0,0 +1,98 @@
+﻿namespace StringCalculator
+{
+    public class StringCalculator
+    {
+        private const int maxNumberToAdd = 1000;
+
+        public int Add(string numbersGiven)
+        {
+            if (string.IsNullOrEmpty(numbersGiven))
+            {
+                return 0;
+            }
+
+            string[] splitNumbersToAdd = SplitStringWithDefaultOrCustomSeperators(numbersGiven);
+
+            ValidateNumbersForNegatives(splitNumbersToAdd);
+
+            string[] splitAndFilteredNumbers = FilterNumbersToSumUnderLimit(splitNumbersToAdd);
+
+            int finalResult = CalculateSplitNumberAddition(splitAndFilteredNumbers);
+
+            return finalResult;
+        }
+
+        private string[] SplitStringWithDefaultOrCustomSeperators(string numbersToSplit)
```

Comment:
This method has too many responsibilities - it is both getting the delimiters and splitting the string with them. See about splitting some of the logic out.